front-end/src/pages/SearchEvents/SearchEvents.jsx

```diff
@@ -38,23 +97,81 @@ function SearchEvents() {
             "start_date": searchDateStart, 
             "end_date": searchDateEnd, 
             "location": searchLocation,
-            searchType: searchTerm
+            [searchType]: searchTerm
         }
         // Calls the getEventDetailsSearch function from EventUtilities to get the events that match the search parameters
         getEventDetailsSearch(allData)
             .then((response) => {
                 setSearchEvents(response)
             })
     }
+    // Gets user locations for rendering events based on user location upon page render
+    const getUserLocations= async () => {
+        const userResponse = await getUserProfile(user);
+        const locations = userResponse.location
+        setUserLocations(locations)
+    };
+
+    useEffect(() => {
+        getUserLocations()
+    }, [userLocations]);
+
+    const getLocalEvents = async () => {
+        const allData = {
+            "type": undefined, 
+            "start_date": undefined, 
+            "end_date": undefined, 
+            "location": userLocations,
+            searchType: undefined
```

Comment:
Is there a reason that you listed all of the variables instead of just location?